### PR TITLE
comparing double with int64 requires a cast

### PR DESCRIPTION
**Pull request by trufae**

### CODE REVIEW

File: quickjs.h [hunk 1]
```diff
@@ -671,7 +671,13 @@ static js_force_inline JSValue JS_NewFloat64(JSContext *ctx, double d)
         uint64_t u;
     } u, t;
     u.d = d;
-    val = (int32_t)d;
+    if (d < INT32_MIN) {
+           val = INT32_MIN;
```
Comment (oleavr) [inline quickjs.h]: Indentation should be 4 spaces (here and below).

**Review by oleavr — CHANGES_REQUESTED** [review]
Thanks!

File: quickjs.h (hunk 1 above)
Comment (trufae) [reply quickjs.h]: fixed

**Comment by trufae** [comment]
also, rebased. feel free to squash or change the commit message

**Pull request merged**